src/Factory.php

```diff
@@ -103,10 +104,22 @@ final public function create(array|callable $attributes = []): Proxy
 
         // filter each attribute to convert proxies and factories to objects
         $mappedAttributes = [];
+        $postPersistCallbacks = [];
         foreach ($attributes as $name => $value) {
             $normalizedAttribute = $this->normalizeAttribute($value, $name);
-            if (self::NULL_VALUE === $normalizedAttribute) {
-                $normalizedAttribute = null;
+
+            if (is_array($normalizedAttribute)) {
+                $postPersistCallbacks = [
+                    ...$postPersistCallbacks,
+                    ...array_values(array_filter($normalizedAttribute, static fn(mixed $attribute): bool => $attribute instanceof PostPersistCallback))
+                ];
+
+                $normalizedAttribute = array_values(array_filter($normalizedAttribute, static fn(mixed $attribute): bool => !$attribute instanceof PostPersistCallback));
+            }
```

Comment:
ok, there is a trick here, sometimes `normalizeAttribute()` will return arrays with some `PostPersistCallback` AND with some proxyfied objects.
So, we need to extract the callbacks on one side, and the regular attributes on the other side.

Maybe to remove this awful thing, we could make `normalizeAttribute()` return some `AttributesCollection`?

Reply:
return type of what? `normalizeAttribute()`?

Reply:
it's `mixed|PostPersistCallback|list<mixed|PostPersistCallback>` I think

Reply:
but... then it's `mixed`  :smile:

Reply:
we return a `list<>` when we have a `X-To-Many`

Reply:
Ok, yeah, that's more complex than I thought...

Reply:
actually here is the full list of what we can return:
- any scalar and any "normal" property
- objects created by a "sub" factory (`X-To-One`  with cascade:persist)
- a `PostPersistCallback` in the case of a `X-To-One` without cascade:persist
- a list of objects (`X-To-Many` with cascade:persist)
- a list of `PostPersistCallback` in the case of a `X-To-Many` without cascade:persist

and this test creates a list of `PostPersistCallback` AND objects created by the factory:
```php
        $category = create(Category::class, [
            'name' => 'bar',
            'posts' => [
                factory(Post::class, ['title' => 'Post A', 'body' => 'body']),
                create(Post::class, ['title' => 'Post B', 'body' => 'body']),
            ],
        ]);
```

Reply:
Damn... so you're thinking an `AttributesCollection` could encapsulate this complexity?

Reply:
The only thing an AttributCollection will do is to hide the filtering process here I guess
But what we'd need is an abstraction and a `PersistenceAttributeCollection`
Don't know if we want such a complexity 🤔

Reply:
Perhaps let's leave out for the context of this PR?

Reply:
For sure, we'll see in the next step. 

But if we create a `PersistenceAtrributesNormalizer` with a method `getPostPersistCallbacks()` (not really sure we willbe able to do this, but that would be useful) we won't need any `AttributesCollection`at all